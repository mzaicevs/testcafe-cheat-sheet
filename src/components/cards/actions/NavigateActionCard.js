import React from "react";
import { actions_navigate } from "../../../snippets";
import CodeCard from "../../../base_components/CodeCard";

const NavigateActionCard = () => (
  <CodeCard
    cardTitle={actions_navigate.title}
    fullDocLink={actions_navigate.fullDocLink}
    codeSnippet={{
      code: actions_navigate.code,
      language: actions_navigate.language,
      example: actions_navigate.example
    }}
    description={actions_navigate.description}
  />
);

export default NavigateActionCard;
