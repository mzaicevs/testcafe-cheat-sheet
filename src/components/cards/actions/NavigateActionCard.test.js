import React from "react";
import { shallow } from "enzyme";
import NavigateActionCard from "./NavigateActionCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("NavigateActionCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<NavigateActionCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
