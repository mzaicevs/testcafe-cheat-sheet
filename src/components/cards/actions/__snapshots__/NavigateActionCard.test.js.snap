// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigateActionCard matches snapshot 1`] = `
<CodeCard
  cardTitle="Navigation - Action"
  codeSnippet={
    Object {
      "code": Array [
        "t.navigateTo( url )",
      ],
      "example": "fixture \`My fixture\`
  .page \`http://www.example.com/\`;

test('Navigate to the main page', async t => {
  await t
      .click('#submit-button')
      .navigateTo('http://devexpress.github.io/testcafe');
});
test('Navigate to local pages', async t => {
  await t
      .navigateTo('file:///user/my-website/index.html')
      .navigateTo('../my-project/index.html');
});",
      "language": "javascript",
    }
  }
  description="Navigates to the specified URL."
  fullDocLink="https://devexpress.github.io/testcafe/documentation/test-api/actions/navigate.html"
/>
`;
